# Lesson 1

We have 8 3 hour lectures
We don't have a book

> Names of the teachers
> * Στραυοποδις
>> He will try and teach us the most simle and comprehensive way
>> We can interact with him at any time
>> It's more so to teach the non biologist
>> He spent six years in the university of Crete
>> Academically he wants  to inspire us, the base is 9 in grades
>> The idea is to have a way out for what you want to choose as a career
> * Bελετζας (just for the technical stuff)
>
> Both come form the biology department

## Principles of Human Genomics

### From next generation sequencing to next generation scociety

In the title of the section they have **highlighted** some of the initials and <u>underlined</u> them. This is to showcase an acronym.

<!-- technical difficulty till now its 12:20 -->
<!-- quick appology -->
<!-- quick majoring of the students -->

> N.
> G.
> S.
Next generation sequenceing
> N.
> G.
> S.
Next generation scociety

He considers sequencing the most innovative technology.

History of the DNA structure.
> 70 Years ago with Watson and Crick
> 50 Years later Watson wanted to convince the U.S. govenrment to start the human genome project
> DNA is a polymer

They collected money for the analysis of the human genomes letters.
> A
> T
> G
> C

This information the bases have is what we will discuss today.
We try to read and enumerate this information in a very complicated way.
> 1939

Huge rpogress has been made since

Sequencing is finding the ATGC letters in the row.

This is because the DNA has repeats.

* Multiplicity
> Is a movie about a guy who clones himself

The different versions of him have different characteristics.

To clone things and to repeat htings is a proble. In nature variation is important.
You don't exist if everythihng is the same.
Data science is based on variation.

If things change space wise or temporal wise, it's something to invest in.

### The human genome project

DNA is comprised of two clones.
> They are complimentary

These are sorted into pairs of bonds.
These bonds stabilize the double helix energetically.

So we have to sequence around 7 billion bases.
With many repeats.
This was a titanic task.

There were two groups 

> The pulic sector (Craig Nernter the most ambitous guy ever)
> The private sector (Bill Nlinton, and the controversy about his secratary)

They tried to combine these two to make one sequence.
they selebrated in may 2000
Aris Patrinos helped this coming together

> Epigenetics mean differences that happend on DNA but not bases changing to one another, but rather from chemical groups that are added to the DNA

In modern biology anything that can be a list of thing scan be define by an **-ome**.
> Genome
> Transcriptome
> Proteome
> Lipidome

We can combine a lot of single units to make an **-ome** sytem.

\[\mathrm{E} = m\times{c^2}\]

Refactored to biology
$\displaystyle \mathrm{E}$ is evolution
$\displaystyle \mathrm{m}$ mutation
$\displaystyle \mathrm{c^2}$ is for cell squared

> Whose genome was completed?

The consensus, the common skeleton

They took a male to keep the *y* chormosome
The *y* is a very interesting chromosome because we thought it had no genes (which is wrong)
(they say they took it form Bill Clinton)

They took immature sperm cells for this.
They broke it up to pieces and each fragment was purified and ***cloned***
This went on recursibly for smaller and smaller fragments.

Trillions of little sequences in the end

Some of the fragments will bind to eachother.
Not a total one a little overlap.
> We must eliminate the randomness.
> So a third was created.

We take the fragments and make a ***contig***.

Seqments of matching pairs that can be sequenced.

He names a couple of countries that invested in the program

> Craig said
> "If I go with this strategy, it will take forever to sequence it all"

So Craig put together genious bio-informaticians.
4 years later he finished the project

Craig sent us a slide of
> A logarithim increase in the number of genomes over time

Some cultures have not sequences their genomes
Information has grown exponentially.

### Groups of genes

Back in 1995 the first bacterial genome was fully sequences.
1996 -> Fungi
1998 -> A worm (C. elegans)
2000 -> Drosophila

At that time we coul not measure the number of genes accuratly.

Now we see a interesting pattern with the gene number and the genome size - cell count.

It seems the number of genes does not reflec the complexity of a oragnism.

Now we know that from one gene its not linear to produce one product.

This leads to the non corralated function we see.

If we had the technology of 1997 it would take 4 million years to complete the human genome project.

You have to think like a computer. Thats your task.

The Wet lab has a lot of impectabilities(?)

*in Silico*. (meaning in simulation)

The techinicians will do the wet lab and we will do the science.

We need to be friends in here to collaborate in order to bring the data together.

"The human pie".

Say we have a new gene. How do I find what it does?
> If it looks like something I know I have a lead

Gene families help sort the data to find similarities in the proteins.
* He goes over some of the families

Half of them seem unknown.

Invent the unknown.

## Describing the reasons we want to sequence DNA

Why do we need this?

> Many diseases can be corralated with genetic traits
> Behavioral ones are harder to isolate to a single gene mutation

### Example of cystic fibrosis

One codon difference 

It's the most common mutation in the human race.

You can diagnose it from 4 years old and after without sequenceing.

We can treat people with gene therapy.

It is important to have the genome sequenced for reasons of prognosis.

### Example of sickle cell anemia

We should just view the; sequeqnces as data

If I know millions of people with a mutation and a clinical conditin I have data.

View it as strictly data.

### Crisp-cas 9

editing cells back to normal state.

## Chromosomes

All choromosomes have genes in specific positions. This is also data because the proximity of the genes to other parts of the chormosome is important to us.

> He gives a list of genes that give pathological phenotypes

Not all mutations are pathogenic. Many of us are polymorphic

Many, for example BRACA gene mutations are polymorphic instead of mutations.

Distinguishing between the polymorphic and the pathologic.

the SRY gene and sex, gender expression.
### What is a chormosome

    Question: about removing the part of the y chromosomes that do nothing
    Answer: When you have a kid, from day 0 till the pregnancy ends we need genes in the area of the y chromosome that do nothing ever after.
    One such thing is the breathing in the amniosack and ever after.

To find out what the parts we dont know about yet is the question. There is no junk.

There are genes that are active for four minuters.

## Genome SNP's

In order to talk about genome snps, we first talk about the tecknology that got us there. In this way we can know the mechanism of delimination as well.

For instance 
> Maxam Gilbert
* Nobel for sequencing
* Too dangerous

> Sanger
* Nobel for sequencing
* Not as dangerous

The Maxam Gilbert method:
He does not explain it properly. You have 4 bands, the two on the side are for two independant nucleotides
the ones in the middle are for either one or another nucleotide. you parse through it and find the contradicitons in the 4 columns.

    Idea
    Take a Sanger graph and make a script that goes to all the positions that are ambiguious and make all the possible sequences.
    It wold make a couple hundred thousand.

> Terminal overlap sequencing
Didnt explain

> On the board DNA example

> AGGCCA
I want to find a way for this to read the sequence. There is a enzyme named DNA polymerase

What Sanger did is he took the double stranded DNA he added temperature and got the opened strands. He added all the nucleotides into the mix with a sample of *dd* nucleotides (slightly different ones that have a molecule that lights up). It just so happenes that *dd* nucleotides will stop the synthesis, thus I get a batch of sequences at random engths that all have one light up nucleotide. Do it enought and the whole sequence will have spots that light up and you can just read it. This produces the complementary strand. Especially cosidering you can design the primers. This also means that to sequence something you have to know something. You should take care to notice the amount of each of the nucleotides.

It's like I said that this (class) is less so for us but more so for the other side
I would have rathered done math rn

# Exam question !!!

    What is the component of the Sanger component that makes the sequenceing stop (typify)?

> Answer
the dideoxy nucleotides

dideoxy nucleotides by definition stop synthesis


DNA is duplex nothing else.
DNA anneals with pH and temperature

- break lasts 15 min

## After the break

### DNA Nanopore technology

* No more thermal denaturatiohn
* Large segments
* Accurate
* No pH annealing
* Cheap
* Convinient
* Works well for DNA and RNA
* No primers
* No waste
* Fast
* electric current and the lipid layer

- It's for small genomes

Invented by Oxford (You could use it for other molecules)

You integrade it in a electrically neutral synthetic membrane. It came from a bacterial protein. You apply a current on either side nad it needs to go through the pore the protein makes. There is a sertain point in the pore where the current can be measured. Every time something blocks the current there is a preturbation in the graph. Due to the diverse structure the nucleotides have they will preterbe in another way. The protein we add is a *genetic* change, the bacterial enzyme called **helicase** will open the double strand and open the DNA. This happens slowly so that the bases can be seen distincly in the graph. The other strand will also be pulled through after the first ends and thus i get a doulbe strand read, they will correct one another.

    Q: Why do we need both strands?
    A: All technology is error prone...

We see a video on the how the nanopore works

> My interpretation:
> Seems to have to do with the receptor and the how the top of it can be adapted to a protein that only has a big enough gap for nucleotides to go through. When something is within the enzyme we get a read, the nucleotides only fir one at a time, thus its liek a type writer that works on its own to read all the nucleotides on the sequence. This also makes it so that the un-annealed sequences can be put througth the sequencing meaning less needs to have happened before the sequencing.

    Idea
    Program that does the antibody recombination in simulation.

## Illumina technology

## Ion torrent sequencing

Video

* Popular
* slower
* expensive
* smallest pH meter

It uses lots of sequences on one large spheres inside pores that, when the sequences create hydrogen bonds, the pH changes. At the bottom of the well there is the smalles pH meter in the world. So we track the amount of the hydrogen bonds we make. This should mean that bonds with 2 and 3 bonds can be separated. It's based on the Sanger sequence but not on the *dd*. This still needs annealing. This needs DNA breaking to happend before becuase the sphere bound DNA is of a sertain length. Each nanosphere will have one fragment that derives from the whole human genome. Millions of these work in parallel. We pcr the same fragment on each sphere. So each sphere has multiple copies of the same sequence, singe stranded. In each cell every 15 seconds I add dna polymerase and dATP but only one nucleotide at a time. The computer does the combining of the different sequences.

## Pyrosequencing

* Denaturing
* Single strand
* Primers
* Pyro Phosphates (PPI)

sulphurilase -> luciferace -> apyrace

It does a traditional sequencing where we add a specific nucleotide one at a time, upon which the reaction will produce a light that will be picked up by a meter, The more of the dATP added we put the stronger the signal. Then we remove the solution from the dna and add a different base. In this we way we know each step the polymerase does. A PyroPhosphate is release (PPI). A bacterial enzyme named ***sulphurilase*** will produce one molecule of ATP. ***Luciferase*** will take the one ATP and make one photon. After this ***apyrase*** is for cleaning the rest of the nucleotides. If more that one photon are made it means that the nucleotide I put is the next 3 positions. No light? ***Apyrace*** clears and no photon is made.

## Next generation sequencing

## Synopsis

For the reactions
* Hydrogen bonds = recognition
* Covalent bonds = stability

# Lesson 2

Online chapters

Smeagol in the slides what da hell. diddy in the slides what da hell.

We discuss the importance of genetic variation which used to be defined as genetic polymorphism

everyone has two eyes one mouth yet still we have so many differences phenotipically.

Polymorphism is single nucleotide changes.

They are very important for disease predisposition. This has a lot to do with the cure we can provide people

## The functional asciciation between a drug response and the genetic variation in the human genome

SNP: For one single position the nucleotide can differ from one person to another

We have almost 1,5 million snp's per genome.

60.000 in Exons. The rest outside. This is around 1 snp per 5000 base pairs.

A link SNP is outside the regulatory and coding section of a gene and have no effect.

Causitive SNP: in the gene or to some degree in the regulatory system.
Then they can also be:
* Non-coding: in a regulatory part | no aminoacid change
* Coding: changes the sequence, these are:
    0. ? Silent, same aminoacid coded essentially non coding
    1. Synonimous: similar aminoacid
    2. Non-Synonymous: different functionally
        Missence: different structure
        Nonsense: Stop codon

SNP's role in disease

We have ascociated a cohourts of snps with specific diseases

These are not mutations they are snps that cna be asciciated with specific diseases and increased risks but not causal chains.

This gets worse when we consider the drugs you can give someone for a target therapy.

We see a atp7a,b exporters and copper to attack cancer cells. cycplatin the protein and neutralisation

Glutatheone sytems are able to change the chemical structure of systatin (not cycplatin). If natural cells can fight the taxic effect of systatin then they will be fine

if ctr1 or ctr2 is expressed systatin cant enter

if a cancer cells expresses alot of atp7a,b then systatin cant stay in the cell.

if sytatin is neutralized byt hte glutatheone systam it wont be functional.

If none of this is true but it causes mutations that are easily repaired then its as if there is no effect.

Kidney transplants and glycoprotein b. 3 conditions: the gene resposible for glycoprotein p production (mdr1): allele snps and differential production. T acrolymous will be able to go into the blood vessil better. then the kidney transplant is accepted.

You should sequence everything before going forward with any medication.

Now we provide medicine from a genetic profile, unlike back in the day when the graft was a littel less determined.

DNA Microarrays, little chips with glass and wells.

You can glue dna strands on glass areas. In a robotc system we can subdivide the wells even further, a microarray with a array in each well.

We watch a video. The idea is to in a defined way imobilise and glue sequences fo my preference on a1, a2, ... in a sertain position on  a disc collection mamking a small librabry of small nucleotides. There is a component that stops the nucleotide from binding onto the glass, we cover everything, then we remove the cap from the unmasked positions, then you add the nucleotide (you just poure over the chip), redo for as many steps.
Why are you doing this, say someone has a infection, waiting 2 to 3 weeks to get a diagnosis is painful. I take some dna immobilise it in the poisitions for all the differetnt diseases i think it could be, then i hynridize and find the one it actually is.

we go over cDNA

A tumor is heterogenious, a small island does not represent the whole cancer

fluresent detection dna chip

We do polymerisation witht taq

# lesson 3

We talk about wet lab. Technical stufff techniques principles

## Souther blot 

Was invented by Southern, he manipulated DNA  and he wanted to find the structure of a gene based in this techinique

Video:
Eppendorfs with dDNA in a buffer are pre loaded and with a pippette you add a enzymes that cut DNA, (restriction enzymes) they digest the DNA. The new take some of the cleaved DNA, and add it to a electrophoresis gel. Small fragments move faster bigger ones move slower. You ideally want to load the product of multiple different restriction enzyme profiles in different wells form different pipettes. The DNA moves throught the gel and based on its end position you can find its size. We dont have a visual yet other than a dye we have added that moves the fastest (it's purpose is to show us when to stop the electrophoresis). You take the gel and add ethidium bromide. Under uv this will glow and my positions will be visible in the gel. I want to now know the sequence of each zone. I make a tank with the gel I denature the DNA in situ (if you add hcl in the buffer for 10 mins the Guanines will be removed and i want this becasue its the biggest and the largest fragments will be easier to see since they move from the top where they are unable to be differentiated). If it's too complex I can't blot well. To make the blot you neutralise in saline, and add a plastic membrane PVDFS that absords the denatured dna on to the membrane. (Why? I can't work with agarose) Now we re hybridize it to something we know about. Once done we add radiactive material to visualize. Now I take whatever i suspect is in the restriction ezyme profile and i hybridize it to see if or not its within the solution i started with. In order to know what part was natured wit hthe radiolabel i use a film that hte radiation will burn onto.

## Cloning

The plasmid with a new gene you want to insert. That old thing.

Applications:
(Watson strand is the 5'3' in that order in the view Crick is the other one)
Plasmids: We go throught the whole plasmid recombination.


Differences wit hpcr

For pcr i need primers so if i dont know the sequence then i cant do pcr however the cloning does not need primers and i can do that first
I can play bakc and fourth. You can freeze repopulate and others for cloning while sequencing is a little stable you only sequence ti after a pcr.

Different vectors

plasmid 5-10kb accomidate small fragments <5000kb
Bacteriophage l 50kb, Linear, accomodate 15000kb
Cosmid, accomidates 40kb 
BAC Bacterial aritificail chormosome

Note to only clone back the cDNA because introns cannot be removed.

Knock in: change something leave tha gene intact
Knock out: change and deactivate the gene
Knock down: Didnt say

RFLPs
restricton fragment length plymorphism
When a polymorphism makes a restriction enzyme sequence in the middle of a sequence that in general does not have that cut.
be careful of false positives in rflps in the sickle cell anemia example
Huntingtons example

## Northern blot Rna hybridisation



## Westeren blot

Antigenifcally cahracterise a protein.

This time we use polyacrilamide not agarsoe remember

Video:
Output regulation of genes in a way, Destroyed cells with all their components in the solution get loaded in the gel after being SDS processed and boiled. This denatures and unfolds the porteins so that they can move based on their weight and not their charge. Let them run. Blot the gel with a higly sensitive antibody assay that recognises the protein of preference. Remember that the loading happens ontop the membrane this time instead of the memebrane on the gel because diffudion would take forever so we add an electric field. Porteins will be renaturated in the second electrophoresis (doubt). Slow blots are better as a result (with methanol). Add bovine serum alvumin to cover non specific sites and the nature with the antibody 1. Come back wash the blot and bind the second anitbody (the antibody antibody with the enzyme).

This helps find meta translational changes like ubuquitination.

Reminder taht resolution limit is a thing and that any band might be more than 1.

After the blot you can take the proteins form a line and cut with tripsin and run MLML liquid chromatography mass spectrometry tandem.
ip western immuno percipitation.

# Lecture 4 missed due to graduation

She uploads her lectures its good

# Lecture 5 

> new dude

Transcriptomics and RNA seq

listing of rna types

RNA seq
* No apriori
* Qualitative and quantitative
* High resolution

0. Extract all rna
1. there is a lot of rRNA in the cell so to get your rna remove it first
2. Convert to cdna

Template switching
adding a synthetic sequence to the end of the primers
improves yield

A new type of graph in phylogenetics that representa time and each individuals life is a specific elongated length that can overlap on their track other familiy tree rules apply
