\documentclass[12pt, a4paper]{article}
\nonstopmode % To make sure that you dont have to press input for each error
\usepackage{graphicx} % Required for inserting images
\graphicspath{{./images}}
\usepackage{pgfplots}
\pgfplotsset{compat=1.18}
\usepackage{mathtools}
\usepackage{fancyhdr}
\usepackage{cancel}
\usepackage[top=1in, left = 1in, right = 1in, bottom=1.2in]{geometry}
\usepackage{listings}
\usepackage{booktabs}
\usepackage{tabularx}
\usepackage{subfig}
\usepackage{hyperref}
\usepackage{float}

% Herlink setup
\hypersetup{
    colorlinks=true,
    linkcolor=blue,
    filecolor=magenta,
    urlcolor=cyan,
    pdftitle={Introduction To Bioinformatics Project},
    pdfpagemode=FullScreen,
}

% For the code blocks
\definecolor{codegreen}{rgb}{0.03,0.5,0.03}
\definecolor{codegray}{rgb}{0.5,0.5,0.5}
\definecolor{codepurple}{rgb}{0.58,0,0.82}
\definecolor{backcolour}{rgb}{0.95,0.95,0.95}

% Code block setup
\lstdefinestyle{mystyle}{
    backgroundcolor=\color{backcolour},
    commentstyle=\color{codegreen},
    keywordstyle=\color{magenta},
    numberstyle=\tiny\color{codegray},
    stringstyle=\color{codepurple},
    basicstyle=\ttfamily\footnotesize,
    breakatwhitespace=false,
    breaklines=true,
    captionpos=b,
    keepspaces=true,
    numbers=left,
    numbersep=5pt,
    showspaces=false,
    showstringspaces=false,
    showtabs=false,
    tabsize=2,
    escapeinside = {(*}{*)}
}
\lstset{style=mystyle}

% My custom headers and margins 
\pagestyle{fancy}
\setlength{\headheight}{44pt}
\setlength{\headsep}{18pt}
\lhead{\includegraphics[scale = 0.2]{~/Documents/masters/bnw unit.png}}
\chead{\quad Data Science and Information Technologies Masterâ€™s
National and Kapodistrian University of Athens}
\rhead{}
\lfoot{}
\cfoot{\thepage}
\rfoot{}

% Start
\begin{document}

% Custom title page
\begin{titlepage}
    \centering
    {\huge \textbf{Project}\par}
    \vspace{0.5cm}
    {\Large \textbf{Name:} Aris Podotas\par}
    \vspace{0.5cm}
    {\large \textbf{University:} National and Kapodistrian University of Athens\par}
    \vspace{0.5cm}
    {\large \textbf{Program:} Data Science and Informaion Technologies\par}
    \vspace{0.5cm}
    {\large \textbf{Specialization:} Bioinformatics - Biomedical Data\par}
    \vspace{0.5cm}
    {\large \textbf{Lesson:} Introduction To Bioinformatics\par}
    \vspace{0.5cm}
    {\large \textbf{Date:} January 2025\par}
    \vspace{0.5cm}
    {\large \textbf{Paper:} Combinatorial assembly of developmental stage-specific enhancers controls gene expression programs during human erythropoiesis\par}
    \vspace{0.5cm}
    {\large \textbf{Link:} https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3477283/\par}
    \tableofcontents
\end{titlepage}

\section{Preface}

The following analysis was done a Linux computer and all the paths will be of the relative file scheme on that computer. This was not done on the virtual machine for there is a desire to apply this analysis on seperate problems in the future and a installation that would be able to do so locally is prefered.
\newline

Along with this report file there will be a $$ file for the script that generates the results, $$ file that has the log of the outputs, all the files used for the analysis (such as the .sra files).

\section{Description of the data}

\subsection{Getting the data}

The following listing shows the part of the $project.sh$ file that fetches the data.

\begin{lstlisting}[language=bash, caption=Downloading the data]
#!/bin/bash

# Chip seq pipleine

# Installing the sra toolkit
wget --output-document sratoolkit.tar.gz https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz
tar -vxzf sratoolkit.tar.gz

# Downloading files
prefetch SRR452931
mv ./SRR452931/SRR452931.sra ./
rm ./SRR452931/
fastq-dump -v --split-3 SRR452931 >> output.log
prefetch SRR524934
mv ./SRR524934/SRR524934.sra ./
rm ./SRR524934/
fastq-dump -v --split-3 SRR524934 >> output.log
prefetch SRR524936
mv ./SRR524936/SRR524936.sra ./
rm ./SRR524936/
fastq-dump -v --split-3 SRR524936 >> output.log
prefetch SRR524939
mv ./SRR524939/SRR524939.sra ./
rm ./SRR524939/
fastq-dump -v --split-3 SRR524939 >> output.log

# Getting the genome
wget -a https://hgdownload.cse.ucsc.edu/goldenpath/hg18/bigZips/hg18.chrom.sizes
wget -a https://hgdownload.cse.ucsc.edu/goldenpath/hg18/bigZips/hg18.fa.gz
gzip -d hg18.fa.gz

bowtie-build hg18.fa hg18
\end{lstlisting}

\section{Protocol}

We should note that the page we downloaded $SRR524939$ from says that the antibody is None, meaning that this is the control file.

\section{Statistics}

\section{References}

\section{Results}

\end{document}

